authors/apps/articles/views.py

```diff
@@ -142,3 +143,122 @@ def put(self, request, slug, *args, **kwargs):
                     serializer.errors,
                     status=400
                 )
+
+
+def get_article(slug):
+    """
+        Returns specific article using slug
+    """
+    article = Article.objects.all().filter(slug=slug).first()
+    if article is None:
+        raise exceptions.NotFound({
+            "message": error_msgs['not_found']
+        }, status.HTTP_404_NOT_FOUND)
+    return article
+
+
+class ShareViaEmail(generics.CreateAPIView):
+    """
+        Share Articles via email
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, slug):
+        """
+            POST /api/v1/<slug>/share/email/
+        """
+        email = request.data['email']
+        get_article(slug)
+
+        if not email:
+            return Response({
+                'message': error_msgs['no_email'],
+            },
+                status=status.HTTP_400_BAD_REQUEST)
+        elif re.search(r"^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$", email) is None:
+            return Response({
+                'message': error_msgs['email_format']
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        username = request.user
+
+        # format the email
+        host = request.get_host()
+        protocol = request.scheme
+        shared_link = protocol + '://' + host + '/api/v1/articles/' + slug
+        subject = "Authors Haven"
+        message = render_to_string(
+            'article_share.html', {
+                'username': str(username).capitalize(),
+                'link': shared_link
+            })
+        to_email = email
+        from_email = os.getenv("DEFAULT_FROM_EMAIL")
+
+        send_mail(
+            subject,
+            message,
+            from_email, [
+                to_email,
+            ],
+            html_message=message,
+            fail_silently=False)
+
+        message = {
+            'message':
+            success_msg['share_success'],
+            'shared_link': shared_link
+        }
+        return Response(message, status=status.HTTP_200_OK)
+
+
+class ShareViaFacebook(generics.CreateAPIView):
```

Comment:
@BoltC0rt3z I noticed that the facebook and twitter share class are similar, you might want to refactor the classes.

Reply:
Thanks for the feedback, I have worked on this